fedora_messaging/cli.py

```diff
@@ -452,3 +458,41 @@ def record(exchange, queue_name, routing_key, app_name, limit, file):
     _consume(
         exchange, queue_name, routing_key, messages_recorder.collect_message, app_name
     )
+
+
+URL_TEMPLATE = "https://apps.fedoraproject.org/datagrepper/id?id={}&is_raw=true"
+
+
+@cli.command()
+@click.argument("message_id")
+@click.option("--datagrep-url", help=_datagreeper_url, default=URL_TEMPLATE)
+def replay(message_id, datagrep_url):
+    """Replay a message from Datagrepper by its message ID"""
+    message_data = _get_message(message_id, datagrep_url)
+    if message_data:
+        try:
+            _publish_message(message_data)
+            click.echo(f"Message with ID {message_id} has been successfully replayed.")
+        except Exception as e:
+            click.echo(f"Failed to replay message: {e}")
+    else:
+        click.echo(f"Could not find message with ID {message_id}")
```

Comment:
Here, it would be more appropriate to raise [click.ClickException](https://click.palletsprojects.com/en/8.1.x/exceptions/), this will instruct click to return with a status code > 0, and would print the error message to stderr.